server/lib/session/session-api.ts

```diff
@@ -46,6 +47,15 @@ export class SessionApi {
 
   @httpGet('/')
   async getCurrentSession(ctx: RouterContext): Promise<ClientSessionInfo> {
+    const {
+      query: { locale },
+    } = validateRequest(ctx, {
+      query: Joi.object<{ date: number; locale?: string }>({
+        date: Joi.number().required(),
+        locale: Joi.string(),
```

Comment:
I think validating these completely is kind of overly complicated and probably not necessary, but you should at least check the length imo. Max length should be 35 characters:
https://stackoverflow.com/a/17863380